components/external_storage/export/src/service.rs

```diff
@@ -0,0 +1,68 @@
+use crate::export::create_storage;
+use external_storage::{ExternalStorage};
+use std::io::{ErrorKind};
+
+use grpcio::{self, *};
+use kvproto::backup as proto;
+use std::collections::HashMap;
+
+type Storages = HashMap<String, Box<dyn ExternalStorage>>;
+
+/// Service handles the RPC messages for the `ExternalStorage` service.
+pub struct Service {
+    storages: Storages,
+}
+
+impl Service {
+    /// Create a new backup service.
+    pub fn new(storages: Storages) -> Service {
+        Service { storages }
+    }
+}
+
+impl proto::ExternalStorage for Service {
+    fn start(
```

Comment:
May be we can  handover the storage preparing works to grpc service itself. The grpc service can read configurations which contains a storage list  from a local file when it starts up.

Reply:
This could work well for many use cases. And I think we can support this pre-loading. However, currently the br tool can specify an arbitrary backend configuration when it requests a backup.

Reply:
The GRPC API has changed a lot now.